### PR TITLE
Add expression and render tests for `within` expression

### PR DESCRIPTION
**Pull request by zmiao**
Render tests and expression tests for https://github.com/mapbox/mapbox-gl-native/pull/16157

### CODE REVIEW

File: test/integration/render-tests/within/paint-icon/style.json [hunk 1]
```diff
@@ -0,0 +1,104 @@
+{
+  "version": 8,
+  "metadata": {
+    "test": {
+      "width": 150,
```
Comment (alexshalamov) [inline test/integration/render-tests/within/paint-icon/style.json]: I think viewport size could be decreased to 64x64?

File: test/integration/expression-tests/within/basic/test.json [hunk 2]
```diff
@@ -0,0 +1,56 @@
+{
+  "expression": ["within", {
+      "type": "Polygon",
+      "coordinates": [[[0, 0], [0, 5], [5, 5], [5, 0], [0, 0]]]
```
Comment (alexshalamov) [inline test/integration/expression-tests/within/basic/test.json]: what about negative tests that would exercise error code path?

Comment (zmiao) [reply test/integration/expression-tests/within/basic/test.json]: Added, thanks!

File: test/integration/render-tests/within/paint-icon/style.json (hunk 1 above)
Comment (zmiao) [reply test/integration/render-tests/within/paint-icon/style.json]: changed.

**Comment by zmiao** [comment]
@mourner  Thanks for the advise! I changed the color to black now.

**Pull request merged**